src/main/scala/gemmini/ReservationStation.scala

```diff
@@ -389,31 +363,38 @@ class ReservationStation[T <: Data : Arithmetic, U <: Data, V <: Data](config: G
   }
 
   // Issue commands which are ready to be issued
-  Seq((ldq, io.issue.ld), (stq, io.issue.st), (exq, io.issue.ex)).foreach { case (q, io) =>
-    val issue_valids = entries.map(e => e.valid && e.bits.ready() && !e.bits.issued && e.bits.q === q)
+  Seq((ldq, io.issue.ld, entries_ld), (exq, io.issue.ex, entries_ex), (stq, io.issue.st, entries_st))
+    .foreach { case (q, io, entries_type) =>
+
+    val issue_valids = entries_type.map(e => e.valid && e.bits.ready() && !e.bits.issued)
     val issue_sel = PriorityEncoderOH(issue_valids)
     val issue_id = OHToUInt(issue_sel)
-    val issue_entry = Mux1H(issue_sel, entries)
+    val global_issue_id = Cat(q.asUInt, issue_id)
```

Comment:
Does this assume that the queue-lengths are powers-of-2?

If so, can you add another assert here? Something like this:

```
assert(isPow2(issue_id.getWidth))
```

Reply:
Addressed earlier